lib/facility.rb

```diff
@@ -1,14 +1,72 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility = {})
+    @name = facility[:name]
+    @address = facility[:address]
+    @phone = facility[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def collect_fees(car)
+    if car.plate_type == :regular
+      @collected_fees += 100
+    elsif car.plate_type == :ev
+      @collected_fees += 200
+    elsif car.plate_type == :antique
+      @collected_fees += 25
+    end
+  end
+
+  def register_vehicle(car)
+    if @services.include?('Vehicle Registration')
+      car.registration_date = Date.today
+      car.plate_type
+      collect_fees(car)
+      @registered_vehicles.push(car)
+    end
+  end
+
+  def administer_written_test(registrant)
```

Comment:
For this method, the written test has to be a service provided and a registrant has to be at least 16 and a registrant has to have a permit. Here you are using the `||` operator and I think you would want to use the `&&` operator that checks the above mentioned conditions. So something like: 

```
if @services.include?('Written Test') && registrant.permit == true && registrant.age >= 16
  registrant.license_data[:written] = true
else
  false
end
```